### PR TITLE
Email notifications

### PR DESCRIPTION
**Pull request by eevelweezel**
Adds the ability to send email notifications via SendGrid (@ZaxR, we should probably discuss credentials).  
The requestee will get a notification when a request is created, and the requestor will get a notification when a requestee accepts a request.  
Also, patches the BuddyRequest unit test so that running the tests won't send notifications.

### CODE REVIEW

**Comment by eevelweezel** [comment]
I didn't see anything in Sendgrid about verifying senders, and messages seem to be going out without delay, but Gmail is flagging everything as Spam.  So, I added an EMAIL_ACCOUNT setting, and changed the notifications to use this address.
As far as processing time goes, I went with Sendgrid because they have a perpetually free tier.  This will work with any SMTP service, though.  I've used Gmail accounts for this, in the past.

**Comment by rifferreinert** [comment]
@eevelweezel looks like there are a few conflicts but it looks good to me

**Comment by eevelweezel** [comment]
Updated.

File: buddy_mentorship/settings/local.py [hunk 1]
```diff
@@ -130,6 +130,7 @@
 # Redirect to home URL after login (Default redirects to /accounts/profile/)
 LOGIN_REDIRECT_URL = "/"
 
+<<<<<<< HEAD
```
Comment (rifferreinert) [inline buddy_mentorship/settings/local.py:133]: Missed this in the merge conflict

File: buddy_mentorship/tests.py [hunk 2]
```diff
@@ -1,6 +1,14 @@
 import datetime as dt
 
-from django.test import TransactionTestCase, TestCase, Client
+from django.test import (
+<<<<<<< HEAD
```
Comment (rifferreinert) [inline buddy_mentorship/tests.py:4]: Here too

File: buddy_mentorship/settings/local.py (hunk 1 above)
Comment (eevelweezel) [reply buddy_mentorship/settings/local.py]: Sorry about that, should be fixed now.

File: buddy_mentorship/tests.py (hunk 2 above)
Comment (eevelweezel) [reply buddy_mentorship/tests.py]: Also this.

**Comment by eevelweezel** [comment]
Fixed the conflicted migration

**Pull request merged**